multitenant: skip SepProcessTenant benchmarks by default

ecwall opened the pull request:
Fixes #97600

The separate process tenant benchmarks can freeze under high parallelism due to default values for limits imposed on separate process tenants. The SepProcessTenant benchmarks are now skipped by default to make running the benchmark suite easier.

Use the `--run-sep-process-tenant` flag to enable these benchmarks.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/100191)
<!-- Reviewable:end -->

ecwall committed:
multitenant: skip SepProcessTenant benchmarks by default

Fixes #97600

The separate process tenant benchmarks can freeze under high parallelism due
to default values for limits imposed on separate process tenants. The
SepProcessTenant benchmarks are now skipped by default to make running the
benchmark suite easier.

Use the `--run-sep-process-tenant` flag to enable these benchmarks.

Release note: None

knz reviewed (approved):
I'm ok with this change. I'd like @stevendanna do provide the final opinion though.

stevendanna reviewed (approved):
This seems OK to me. 

One question I have is if they still freeze when we put in an absurdly high limit using `crdb_internal.update_tenant_resource_limits`?  If so, that does sort of imply some problem with the costcontroller infrastructure that we might want to file a bug for.

ecwall commented:
bors r=knz,stevendanna
